frontend/model/state.js

```diff
@@ -329,6 +335,235 @@ const getters = {
       return profile.displayName || username
     }
   },
+  // used with graphs like those in the dashboard and in the income details modal
+  groupIncomeDistribution (state, getters) {
+    return groupIncomeDistribution({ state, getters, monthstamp: currentMonthstamp() })
+  },
+  // adjusted version of groupIncomeDistribution, used by the payments system
+  groupIncomeAdjustedDistribution (state, getters) {
+    return getters.groupIncomeAdjustedDistributionForMonth(currentMonthstamp())
+  },
+  groupIncomeAdjustedDistributionForMonth (state, getters) {
+    return monthstamp => groupIncomeDistribution({ state, getters, monthstamp, adjusted: true })
+  },
+  ourPayments (state, getters) {
+    const currency = currencies[getters.groupSettings.mincomeCurrency]
+    const ourUsername = getters.ourUsername
+    const cMonthstamp = currentMonthstamp()
+    const pDate = dateFromMonthstamp(cMonthstamp)
+    const dueIn = lastDayOfMonth(pDate)
+    const monthlyPayments = getters.groupMonthlyPayments
+    const allPayments = getters.currentGroupState.payments
+
+    const sent = (() => {
+      const payments = []
+      // QUESTION: Is this only from this month? What about the previous payments from months ago?
+      //           Same question for received.
+      for (const monthstamp of Object.keys(monthlyPayments).sort()) {
+        const { paymentsFrom } = monthlyPayments[monthstamp]
+
+        if (!paymentsFrom) { continue }
```

Comment:
Am not a fan of this `if (!paymentsFrom) { continue }` style personally... prefer how it was before

Reply:
How was it before? I'm not understanding what's wrong...

Reply:
Look at the previous code that was deleted in Payments.vue.  e.g.

```js
    paymentsSent () {
      const monthlyPayments = this.groupMonthlyPayments
      const payments = []
      // TODO: support sort direction
      for (const monthstamp of Object.keys(monthlyPayments).sort()) {
        const { paymentsFrom } = monthlyPayments[monthstamp]
        if (paymentsFrom) {
          for (const toUser in paymentsFrom[this.ourUsername]) {
            for (const paymentHash of paymentsFrom[this.ourUsername][toUser]) {
              const { data, meta } = this.currentGroupState.payments[paymentHash]
              const payment = {
                hash: paymentHash,
                data,
                meta,
```